### PR TITLE
Improve styling of password reset email

### PR DESCRIPTION
**Pull request by int-y1**
Make the email fit DMOJ's theme. Before and after:

Capture

This email displays correctly on Outlook, too.
Note: Buttons are impossible to style correctly on Outlook, so I used a link instead.

### CODE REVIEW

File: templates/registration/password_reset_email.html [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
+<div style="border:2px solid #337ab7;margin:4px 0;"><div style="background:#fafafa;border:12px solid #fafafa;font-family:segoe ui,lucida grande,Arial,sans-serif;font-size:14px;">
+<strong>{{ _("Forgot your password on the %(site_name)s? Don't worry!", site_name=site_name) }}</strong><br><br>
 {{ _('To reset the password for your account "%(username)s", click the button below.', username=user.get_username()) }}
-<p align="center">
-<a href="{{ protocol }}://{{ domain }}{{ url('password_reset_confirm', uidb64=uid, token=token) }}" style="cursor: pointer;display:block;text-align: center;padding: 4px 2px 5px;color: white;border: 1px solid #666;border-radius: 1px;background: #2980b9;background: linear-gradient(180deg, #00aee0, #2980b9);text-decoration: none;line-height:2em;font-size:1em;width:12em;">{{ _('Reset Password') }}</a>
+<p style="text-align:center;">
+<a href="{{ protocol }}://{{ domain }}{{ url('password_reset_confirm', uidb64=uid, token=token) }}" style="background:#337ab7;padding:6px 18px;border:1px solid #245580;color:#fff;text-decoration:none;">{{ _('Reset Password') }}</a>
```
Comment (Xyene) [inline templates/registration/password_reset_email.html:9]: We have `border-radius: 4px;` on this button in prod, does that work in emails?

Comment (Xyene) [reply templates/registration/password_reset_email.html:9]: Prod also uses `background: linear-gradient(to bottom,#337ab7 0,#265a88 100%) repeat-x;` if that's supported.

Comment (int-y1) [reply templates/registration/password_reset_email.html]: I replaced the button with a link, because Outlook hates css.

**Review by Xyene — APPROVED** [review]
LGTM, thanks!

**Pull request merged**